app/components/todo-item.js

```diff
@@ -0,0 +1,65 @@
+import Component from '@glimmer/component';
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { service } from '@ember/service';
+import { action, get } from '@ember/object';
+export default class TodoItemComponent extends Component {
+    @service todo;
+    @service router;
+
+    changedHeader = '';
+    changedDescription = '';
+
+    @action modifyFirstName(val) {
+        this.changedHeader = val;
```

Comment:
Please use `ember-set-helper` in the template. 
https://github.com/pzuraq/ember-set-helper